android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/ConnectionHealth.kt

```diff
@@ -242,11 +246,18 @@ class ConnectionHealth(service: VPNService) {
             if (fallbackServerIsReachable) {
                 recordMetrics(ConnectionStability.Unstable)
 
+                if (LocalDateTime.now() < nextPossibleServerSwitch) {
+                    Log.i(TAG, "Want to switch servers, but it has not been enough time since last server switch")
+                    taskDone()
+                    return@Runnable
+                }
+
                 Log.i(TAG, "Switch to fallback VPN server")
                 // We the server is online but the connection broke up, let's rest it

```

Comment:
This comment seems badly formatted it would be nice to touch it up while we're here.

Reply:
Thanks, updated